src/solver/expressions/include/antares/solver/expressions/visitors/NodeVisitor.h

```diff
@@ -49,42 +75,43 @@ class NodeVisitor
     /**
      * Dispatches a node to an appropriate visitor function based on its type.
      *
-     * Iterates through a list of visitor functions, attempting to call each one until a successful
-     * visit is made.
+     * This method uses a map that associates node types
+     * with their corresponding visitor functions. It attempts to find the visitor function
+     * for the provided `node`. If a match is found, the corresponding
+     * visitor function is called with the node, and any
+     * additional arguments (`args...`).
      *
      * @param node A reference to the Node object to be visited.
      * @param args Variadic template arguments to be passed to the visitor functions.
-     * @return An optional value of type `R` representing the result of the visit.
-     *         If no suitable visitor function is found, an error
-     * message is logged.
+     * @return The return value of the visitor function.
+     *
      */
     R dispatch(const Node& node, Args... args)
     {
-        using FunctionT = std::optional<R> (*)(const Node&, NodeVisitor<R, Args...>&, Args... args);
-        static const std::vector<FunctionT> allNodeVisitList{
-          &tryVisit<R, NodeVisitor<R, Args...>, AddNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, SubtractionNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, MultiplicationNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, DivisionNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, EqualNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, LessThanOrEqualNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, GreaterThanOrEqualNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, NegationNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, ParameterNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, VariableNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, LiteralNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, PortFieldNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, ComponentVariableNode>,
-          &tryVisit<R, NodeVisitor<R, Args...>, ComponentParameterNode>};
-        for (auto f: allNodeVisitList)
+        const static auto nodeDispatchFunctions = NodeDispatchFunctionsProvider<R, Args...>::
```

Comment:
**NodeDispatchFunctionsProvider** could be renamed **NodeVisitsProvider** more in related the underlying design pattern.

Same for **nodeDispatchFunctions** : ==> **nodeVisitList** ?